### ISSUE TITLE
Respect Flow

### ISSUE THREAD

**Issue by s-kashif** [hovancik/stretchly#244]
Sometimes we get in the "Flow" and would like to be not disturbed for a while 
(Some way to disable 3-4 future microbreaks)

:)

**Comment by hovancik** [comment]
@s-kashif what settings do you use? maybe setting longer time between micro/breaks could solve your issue? I  general I would prefer using those settings to solve it

@lundibundi that could work, but I am afraid that it would make menu too complicated. We would need to have menu to disable breaks, disable micorobreaks, disable both and logic to handle all posible options... not sure if it's worth to implement and maintain such a functionality

**Comment by s-kashif** [comment]
@hovancik I'm a pretty distracted fellow 😄 so I put 10 sec micro breaks every 5 min and then 3 min break every 30 min.
These super quick micro-breaks are great at breaking procrastination but also get in the way when I'm actually focused.

I would suggest having user customizable profiles like flow-profile, distracted-profile with some shortcuts to quickly switch b/w them.
eg. 
flow-profile -> no micro + 25 min break (Something like CTRL+1 to switch)
distracted-profile -> 5 min micro + 30 min break (Something like CTRL+2 to switch)

**Comment by l-4** [comment]
Becoming an interruption is the main danger for Stretchly in my experience. I have yo-yoed, switching it off intermittently for this reason.

I came here to suggest some sort of "soft warning" or pre-notification before Stretchly pops up on screen. I had imagined the screen pulsating gently. This way one would be aware that shortly an interruption would happen and would have a chance to prepare for it, rather than being rudely interrupted while typing, or reading, or something else.

Reading this topic has given me another idea. Making a complex set of on and off switches or any manual switch defeats the point of having them in the first place - flow just happens, it is not possible to plan it. Is there a way for Stretchly to automatically detect when a user in a state of flow? Perhaps when only one program has been in focus for a while, or only one type of input is in use.

Failing that, perhaps a manual hotkey to activate/deactivate all interruptions could be used during flow with the minimum disruption.

Very keen to help in any way I can. I see Stretchly as a vital addition to my workstation and a valuable contributor to my physical health.
